code/workspaces/infrastructure-api/src/controllers/stackController.js

```diff
@@ -113,10 +114,23 @@ const createStackValidator = [
       }
       return true;
     }),
+  check('visible', 'visible must be specified for sites')
+    .custom((value, { req }) => {
+      if (indexOf(Stacks.getNamesByCategory(PUBLISH), req.body.type) > -1) {
+        return [visibility.PRIVATE, visibility.PROJECT, visibility.PUBLIC].includes(req.body.visible);
```

Comment:
Might be worth doing this as 
```js
visibility.values().includes(req.body.visible)
``` 
or I think there is a function in `StackEnums` called something like `getEnumValues` which you could use to do 
```js
getEnumValues(visibility).includes(req.body.visible)
```

That way, if the enum is updated this won't need to be.